ci/utils/pr_test_summary.py

```
@@ -0,0 +1,220 @@
+                None,
+            )
+
+
+def _api(path, token, method, payload):
+    req = urllib.request.Request(
+        f"https://api.github.com{path}",
+        data=json.dumps(payload).encode(),
+        method=method,
+        headers={**_headers(token), "Content-Type": "application/json"},
+    )
+    with urllib.request.urlopen(req) as resp:
```

Ruff (0.15.18)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)